Additional features

stevenschemers opened the pull request:
Added lots of quality of life improvements. Based on [pull request #2](https://github.com/yonilerner/bakkesmod-playlist-rank-viewer/pull/2).

**general features:**
- added tier & division data to string
- added tier-based highlights to text rows

**Settings update:**
- toggle for rank 
- toggle for mmr 
- toggle for games played
- toggle for tier highlighting
- x & y positioning sliders
- scaling slider
- background opacity slider

stevenschemers committed:
Fix for latest Bakkesmod update

stevenschemers committed:
renamed ONES to SOLO

stevenschemers committed:
updated relative path for bakkes_patchplugin.py

stevenschemers committed:
added additional player stats

stevenschemers committed:
added quality of life features

stevenschemers commented:
I and a bunch of my friends really enjoy this plugin and we thought these features might help add a bit more value to it! I hope you like these ideas and merge them in.

stevenschemers committed:
increased version to 1.3

stevenschemers committed:
recompiled with new version #

yonilerner commented on bakkesmod-playlist-rank-viewer/RankAndPlaylistData.h:
```diff
@@ -0,0 +1,74 @@
+
+array<PLAYLIST, 9> playlistsToCheck = { UNRANKED, SOLO, TWOS, THREES, HOOPS, RUMBLE, DROPSHOT, SNOWDAY, TOURNAMENT };
+
+map<RANK, RankInfo> RankTiers = {
```
Should we delete this version of the map?

stevenschemers replied:
I was considering it, but I didn't want to go too crazy so you can still have some options. Totally up to you. As of right now, it's not being used in the component but does act as a full-length list of the tiers.

stevenschemers replied:
I reduced these 2 maps into a single map in my last commit. I think this makes a lot more sense since we can just store both versions of the string in the same map

yonilerner replied:
I like it

yonilerner commented on bakkesmod-playlist-rank-viewer/pch.h:
```diff
@@ -0,0 +1,9 @@
+#pragma once
```
What does this file do? What does pch mean?

stevenschemers replied:
pch is a precompiled header in c++. I'm not usually a c++ programmer but I've seen people use them as sort of the primary include, intended to prevent multi-line copy/pastes I believe. I felt this would help make it a bit easier to grow without cluttering up the component header too much. We can always do something different if you prefer.

stevenschemers replied:
I actually think this file might be overkill looking at it again. I think when I added this I intended to separate a bit more of the code but it didn't end up needing all that. If it suits you I might revert this change.

yonilerner replied:
Heh im not a C++ programmer so Ill accept your judgement blindly on this :)

yonilerner commented on bakkesmod-playlist-rank-viewer/bakkesmod-playlist-rank-viewer.cpp:
```diff
@@ -76,47 +98,123 @@ void PlaylistRankViewer::timeoutCallback(GameWrapper* _) {
+	};
+	Vector2 sizeBox = {
+		int(255 * PRV_scale),
+		int(int(35 + (20 * playlistsToCheck.size())) * PRV_scale)
```
Could we change 35 -> 40? Its cut a little close for my tastes

stevenschemers replied:
Of course!

yonilerner reviewed (commented):
Im really glad yall are enjoying it! I love this PR thank you so much. I just had a couple small questions/comments. Also I think this needs a rebase now that I merged #2.

yonilerner commented:
Oh also, would you be willing to update the README?

stevenschemers commented:
> Oh also, would you be willing to update the README?

sure thing!

stevenschemers committed:
increased bottom padding on canvas box

stevenschemers committed:
fixed namespacing and dropped pch.h

stevenschemers committed:
Update README.md

updated readme with new features

stevenschemers commented:
@yonilerner I've updated the readme and added the padding to the bottom of the canvas as well as removed the pch.h header. LMK if there are any other changes you'd like to see. 

Thanks again!

stevenschemers committed:
Update README.md

stevenschemers committed:
Update README.md

fixed broken setting screenshot url

stevenschemers committed:
reduced rankTier maps to single map with abbvr data included

stevenschemers committed:
Merge branch 'additional-features' of github.com:stevenschemers/bakkesmod-playlist-rank-viewer into additional-features

yonilerner commented:
@stevenschemers If you wanna add me on Discord Im Yoni#0001
